### PR TITLE
Support ALTER VITESS_MIGRATION ... CLEANUP statement

### PR DESCRIPTION
**Pull request by shlomi-noach**
Followup to #9158  (9158 is unmerged yet, this PR extends it)
Tracking: #6926

This PR offers a new statement:
```sql
alter vitess_migration '9748c3b7_7fdb_11eb_ac2c_f875a4d24e90' cleanup
```

Normally, when a migration is complete, it takes `24h` (or however `-retain_online_ddl_tables` is configured) until the artifacts are garbage collected and removed. 

With this new SQL statement we can tell vitess that a migration's artifacts are good to garbage collect _now_. The statement itself does not clean the artifacts, and returns immediately. It merely sets a small (negative) value to `retain_artifacts_seconds` (see #9158) so that the next poll interval, typically a matter of seconds, can pick up on those artifacts and begin the garbage collection process.

Notice that once a migration's artifacts are cleaned up, the migration may not be eligible for [revert](https://vitess.io/docs/user-guides/schema-changes/revertible-migrations/).

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

Docs are required.

### CODE REVIEW

**Comment by shlomi-noach** [comment]
ready for review

**Review by GuptaManan100 — APPROVED** [review]
Parser changes look good to me! 🏆

File: doc/releasenotes/13_0_0_summary.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+## Incompatible Changes
+
+## Syntax changes
+
+### alter vitess_migration ... cleanup
+A new query is supported:
+```sql
+alter vitess_migration '9748c3b7_7fdb_11eb_ac2c_f875a4d24e90' cleanup
+```
+This query tells Vitess that a migration's artifacts are good to be cleaned up asap. This allows Vitess to free disk resources sooner. As reminder, once a migration's artifacts are cleaned up, the migration is no
```
Comment (deepthi) [inline doc/releasenotes/13_0_0_summary.md:13]:
```suggestion
This query tells Vitess that a migration's artifacts are good to be cleaned up asap. This allows Vitess to free disk resources sooner. As a reminder, once a migration's artifacts are cleaned up, the migration is no
```

**Review by deepthi — APPROVED** [review]
Non-parser changes look good to me.
Minor suggestion on release notes, otherwise :+1:

File: doc/releasenotes/13_0_0_summary.md (hunk 1 above)
Comment (shlomi-noach) [reply doc/releasenotes/13_0_0_summary.md]: fixed

**Comment by shlomi-noach** [comment]
I resolved conflicts with `main`, which merged in the new linter checks. This PR therefore also modifies (minorly) some unrelated code that has to pass new linting.

**Comment by shlomi-noach** [comment]
Documentation PR: https://github.com/vitessio/website/pull/886

**Pull request merged**